src/Persistent/Implementation/AcidState.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE ImpredicativeTypes          #-}
+{-# LANGUAGE OverloadedStrings           #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TemplateHaskell             #-}
+{-# LANGUAGE TypeFamilies                #-}
+{-# LANGUAGE TypeOperators               #-}
+
+{-# OPTIONS_GHC -Wall -Werror #-}
+
+module Persistent.Implementation.AcidState
+    ( Persist
+    , mkRunPersist
+    )
+where
+
+import Control.Lens
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad.Reader (ask)
+import Control.Monad.State (get, put)
+import Data.Acid (AcidState, Query, Update, makeAcidic, openLocalState, query, update)
+import Control.Monad.Trans.Reader (ReaderT(ReaderT), runReaderT)
+import Servant.Server ((:~>)(Nat))
+
+import Persistent.Api
+import Types
+
+-- FIXME: Remove
+import Test.QuickCheck (generate)
+
+newtype Persist a = Persist (ReaderT (AcidState AulaData) IO a)
+  deriving (Functor, Applicative, Monad)
+
+persistIO :: IO a -> Persist a
+persistIO = Persist . liftIO
+
+instance GenArbitrary Persist where
+    genGen = persistIO . generate
+
+-- TODO: call @closeAcidState db@ somewhere
```

Comment:
We need to catch SIGINT, which is a ghc exception.  I remember we've done it in thentos before moving away from acid-state.  if you look at june 2015 there, you might find it.